build_runner/lib/src/build_script_generate/build_script_generate.dart

```diff
@@ -153,3 +146,24 @@ Expression _applyBuilder(BuilderDefinition definition, Expression toExpression,
           .toList()),
       toExpression,
     ], namedArgs);
+
+// TODO - graph argument should be removed once `toRoot()` is a supported
+// PackageFilter
+Expression _findToExpression(
+    BuilderDefinition definition, PackageGraph packageGraph) {
+  switch (definition.autoApply) {
+    case AutoApply.none:
+      return refer('toNoneByDefault', 'package:build_runner/build_runner.dart')
+          .call([]);
+    case AutoApply.dependents:
+      return refer('toDependentsOf', 'package:build_runner/build_runner.dart')
+          .call([literalString(definition.package)]);
+    case AutoApply.allPackages:
+      return refer('toAllPackages', 'package:build_runner/build_runner.dart')
+          .call([]);
+    case AutoApply.rootPackage:
+      return refer('toPackage', 'package:build_runner/build_runner.dart')
+          .call([literalString(packageGraph.root.name)]);
+  }
+  throw 'Unreachable';
```

Comment:
typically I prefer to handle this by throwing a more specific error in the `default` case... I can see an argument for this as well though since it allows us to get an analyzer error if we are missing one of the values. I think its worth throwing a `StateError` with a better message still though.

Reply:
ArgumentError is pretty good: